kernel/packages/scene-system/stateful-scene/SceneStateDefinition.ts

```diff
@@ -0,0 +1,84 @@
+import { CLASS_ID } from "decentraland-ecs/src";
+import { Component, ComponentData, ComponentId, EntityId, StateContainer } from "./types";
+
+export class SceneStateDefinition implements StateContainer {
+
+  private readonly entities: Map<EntityId, Map<ComponentId, ComponentData>> = new Map()
+
+  addEntity(entityId: EntityId, components?: Component[]): void {
+    const componentMap: Map<ComponentId, ComponentData> = new Map((components ?? []).map(({ id, data }) => [id, data]))
+    this.entities.set(entityId, componentMap)
+  }
+
+  removeEntity(entityId: EntityId): void {
+    this.entities.delete(entityId)
+  }
+
+  setComponent(entityId: EntityId, componentId: ComponentId, data: ComponentData): void {
+    this.entities.get(entityId)?.set(componentId, data)
+  }
+
+  removeComponent(entityId: EntityId, componentId: ComponentId): void {
+    this.entities.get(entityId)?.delete(componentId)
+  }
+
+  sendStateTo(container: StateContainer) {
+    for (const [entityId, components] of this.entities.entries()) {
+      const mappedComponents = Array.from(components.entries())
+        .map(([id, data]) => ({ id, data }))
+      container.addEntity(entityId, mappedComponents)
+    }
+  }
+
+  toStorableFormat(): any {
```

Comment:
Shouldn't we maybe separate this to another module? Something like `StateDefinitionSerializer` or similar.

I think it would make more sense, and would also enable us to separate the concerns of having an scene state and being able to store it somehow.